typescript/ica/contracts/OwnableMulticall.sol

```diff
@@ -31,4 +31,30 @@ contract OwnableMulticall is OwnableUpgradeable {
             }
         }
     }
+
+    function _call(Call[] memory calls, bytes[] memory callbacks)
+        internal
+        returns (bytes[] memory resolveCalls)
+    {
+        resolveCalls = new bytes[](callbacks.length);
+        for (uint256 i = 0; i < calls.length; i++) {
+            (bool success, bytes memory returnData) = calls[i].to.call(
+                calls[i].data
+            );
+            require(success, "Multicall: call failed");
+            resolveCalls[i] = bytes.concat(callbacks[i], returnData);
+        }
+    }
+
+    // TODO: deduplicate
+    function proxyCallBatch(address to, bytes[] memory calls) internal {
```

Comment:
Probably name this something differently? Is it not easy to just construct `Call[]` from `bytes[]` with `to` as the `Call.to` field and then reuse `proxyCalls`? Like pass `sender` to `_call` and return `Call[]` instead of `bytes[] resolveCalls`

Reply:
this saves a lot on gas